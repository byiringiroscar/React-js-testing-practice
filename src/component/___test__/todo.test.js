import { render, screen, cleanup } from "@testing-library/react";
import Todo from "../todo";

test('should render non-completed todo', () => {
    const todo = {id: 1, title: 'WASH DISHES', completed: false};
    render(<Todo todo={todo} />);
    const todoElement = screen.getByTestId('todo-1');
    expect(todoElement).toBeInTheDocument();
    expect(todoElement).toHaveTextContent('WASH DISHES');
});

test('should render completed todo', () => {
    const todo = {id: 2, title: 'wash car', completed: true};
    render(<Todo todo={todo} />);
    const todoElement = screen.getByTestId('todo-2');
    expect(todoElement).toBeInTheDocument();
    expect(todoElement).toHaveTextContent('wash car');
});