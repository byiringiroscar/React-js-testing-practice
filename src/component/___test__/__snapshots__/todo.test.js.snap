// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<h1
  data-testid="todo-1"
>
  <h1>
    WASH DISHES
  </h1>
</h1>
`;
